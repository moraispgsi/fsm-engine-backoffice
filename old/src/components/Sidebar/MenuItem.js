/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';
import tinycolor from 'tinycolor2';

import {
  fontFamilyBase,
  fontSizeBase,
  lineHeightBase,
  fontWeightBase,
  sidebarWidth,
  sidebarMiniWidth,
  screenSmMin,
  lightBlue,
  green,
  yellow,
  red,
  aqua,
} from '../../styles/variables';

const Label = styled.div`
  /* shared */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: ${fontFamilyBase};
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;

  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */

  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  display: inline;
  padding: .2em .6em .3em .6em;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: .25em;
  float: right!important;
  color: #fff;
  margin-right: 5px;

  /* ----- color ----- */
  background-color: ${(props) => {
    switch (props.type) {
      case 'primary':
        return lightBlue;
      case 'success':
        return green;
      case 'danger':
        return red;
      case 'warning':
        return yellow;
      case 'information':
        return aqua;
      default:
        return lightBlue;
    }
  }};

  /* ----- collapse ----- */
  ${props => props.collapse && `
    display: ${props.hover ? 'block' : 'none'};
    float: right;
  `}
`;

const RightSpan = styled.span`
  position: absolute;
  right: 10px;
  top: 50%;
  margin-top: -7px;

  color: ${props => props.theme.sidebarLinkColor || '#fff'};

  /* ----- hover ----- */
  ${props => (props.collapse && props.hover && !props.level) && `
    display: block;
    left: ${parseInt(sidebarWidth, 10) - 30}px;
    top: 10px!important;
    margin-left: -3px;
    padding: 12px 5px 12px 20px;
    float: right;
    overflow-x: visible;
  `}
`;

const StyledTitle = styled.span`
  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */

  /* collapse */
  ${props => (!props.level && props.collapse) &&
    `@media (min-width: ${screenSmMin}) {
      border-top-right-radius: 4px;
      ${(props.hover ?
        `
          display: block !important;
          position: absolute;
          width: ${parseInt(sidebarWidth, 10) - parseInt(sidebarMiniWidth, 10)}px;
          left: ${sidebarMiniWidth};
          top: 0;
          margin-left: -3px;
          padding: 12px 5px 12px 20px;
          background-color: inherit;
        ` :
        `
          display: none !important;
          -webkit-transform: translateZ(0);
        `
      )}
    }`
  }
`;

const StyledRightIcon = styled.i`
  width: auto;
  height: auto;
  padding: 0;
  margin-right: 10px;
  margin-left: .3em;
  float: right;

  ${props => props.open && `
    -ms-transform: rotate(-90deg);
    -webkit-transform: rotate(-90deg);
    transform: rotate(-90deg);
  `}

  /* ----- collapse ----- */
  ${props => (props.collapse && !props.hover) && 'display: none !important;'};
`;

const StyledLeftIcon = styled.i`
  width: 20px;

  /* ----- color ----- */
  ${(props) => {
    switch (props.color) {
      case 'primary':
        return `color: ${lightBlue};`;
      case 'success':
        return `color: ${green};`;
      case 'danger':
        return `color: ${red};`;
      case 'warning':
        return `color: ${yellow};`;
      case 'information':
        return `color: ${aqua};`;
      default: {
        const c = tinycolor(props.color);
        return c.isValid() ?
          `color: ${c.toString()};` :
          null;
      }
    }
  }};
`;

const StyledLink = styled.a`
  cursor: auto;
  text-decoration: none !important;
  display: block;
  padding: ${props => (!props.level ? '12px 5px 12px 15px' : '5px 5px 5px 15px')};
  position: relative;
  margin: 0;
  font-size: ${fontSizeBase};
  background-color: transparent;

  &:active,
  &:focus,
  &:hover {
    outline: none !important;
    text-decoration: none !important;
  }

  /* ----- theme ----- */
  /* text color - different in submenu */
  color: ${props =>
    (props.level ? props.theme.sidebarSubmenuLinkColor : props.theme.sidebarLinkColor)} !important;
  /* left border - none in submenu */
  border-left: ${props => (!props.level ? props.theme.sidebarLinkBorderLeft : 'none')};
  /* transition */
  ${props => (props.theme.sidebarLinkTransition && `
    -webkit-transition: ${props.theme.sidebarLinkTransition};
    -moz-transition: ${props.theme.sidebarLinkTransition};
    -o-transition: ${props.theme.sidebarLinkTransition};
    transition: ${props.theme.sidebarLinkTransition};
  `)}
  /* font weight */
  ${props => (props.theme.sidebarLinkFontWeight && `
    font-weight: ${props.theme.sidebarLinkFontWeight};
  `)}

  /* ----- active or hover properties ----- */
  ${props => (props.active || props.hover) && `
    color: ${props.theme.sidebarLinkActiveColor || '#fff'} !important;
    background: ${!props.level && props.theme.sidebarLinkActiveBg};
    border-left-color: ${props.theme.sidebarLinkActiveBorderLeftColor || '#3c8dbc'};
    font-weight: ${props.theme.sidebarLinkActiveFontWeight || 'normal'};
    text-decoration: none !important;
  `}
`;

const StyledSubMenu = styled.ul`
  display: ${props => (props.open ? 'block' : 'none')};
  list-style: none;
  padding: 0 0 0 ${props => ((props.level + 1) * 5)}px;

  /* ----- theme ----- */
  margin: ${props => props.theme.sidebarSubmenuMargin || '0'};
  background-color: ${props => props.theme.sidebarSubmenuBg || '#222d32'};

  /* ----- collapse ----- */
  ${props => props.collapse &&
    `@media (min-width: ${screenSmMin}) {
      ${props.theme.sidebarSubmenuLgBorderLeft &&
        `border-left: ${props.theme.sidebarSubmenuLgBorderLeft}`}
    }`
  }
  ${props => (props.collapse && props.open && !props.level) &&
    `@media (min-width: ${screenSmMin}) {
      position: absolute;
      width: ${parseInt(sidebarWidth, 10) - 50}px;
      left: ${sidebarMiniWidth};
      top: 44px;
      margin-left: 0;
      border-bottom-right-radius: 4px;
      ${(props.theme.sidebarSubmenuLgBorderLeft) &&
        `border-left: ${props.theme.sidebarSubmenuLgBorderLeft}`}
    }`
  }
`;

const StyledMenuItem = styled.li`
  /* shared */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: ${fontFamilyBase};
  font-weight: ${fontWeightBase};
  font-size: ${fontSizeBase};
  line-height: ${lineHeightBase};
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;

  list-style: none;
  position: relative;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  display: list-item;
  text-align: -webkit-match-parent;
`;

const renderChildren = (children, sidebarCollapse, level, hover) => {
  const newLevel = level + 1;
  return React.Children.map(children, (child, index) => (
    React.cloneElement(child, {
      key: index.toString(),
      collapse: sidebarCollapse,
      level: newLevel,
      parentHover: hover,
    })
  ));
};

const renderLabels = (labels, collapse, hover) => (
  labels.map((l) => {
    if (l.key && l.type && l.text) {
      if (l.key === 1) {
        return (<Label
          collapse={collapse}
          hover={hover}
          key={l.key.toString()}
          type={l.type}
        >
          {l.text}
        </Label>);
      } else if (!collapse) {
        return (<Label
          collapse={collapse}
          hover={hover}
          key={l.key.toString()}
          type={l.type}
        >
          {l.text}
        </Label>);
      }
    }
    return null;
  })
);

class MenuItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      hover: false,
    };
    this._toggleMenu = this.toggleMenu.bind(this);
    this._toggleHover = this.toggleHover.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    (nextProps.collapse !== this.props.collapse) &&
      this.setState({ open: false });
  }

  toggleMenu() {
    this.props.children && this.setState({ open: !this.state.open });
  }

  toggleHover(state) { this.setState({ hover: state }); }

  render() {
    return (
      <StyledMenuItem
        collapse={this.props.collapse}
        hover={this.props.parentHover}
        level={this.props.level}
        onMouseLeave={() => this._toggleHover(false)}
      >
        <StyledLink
          active={this.props.active}
          collapse={this.props.collapse}
          hover={this.state.hover}
          href={(this.props.children || this.props.onClick) ?
            null : this.props.href}
          level={this.props.level}
          onClick={this.props.children ?
            this._toggleMenu : this.props.onClick}
          onMouseEnter={() => this._toggleHover(true)}
        >
          <StyledLeftIcon
            className={`fa ${this.props.icon.className || 'fa-circle-o'}`}
            color={this.props.icon.color || 'none'}
          />
          <StyledTitle
            collapse={this.props.collapse}
            hover={this.state.hover}
            level={this.props.level}
          >
            {this.props.title}
          </StyledTitle>
          <RightSpan
            collapse={this.props.collapse}
            hover={this.state.hover}
            level={this.props.level}
          >
            {(this.props.labels ?
              renderLabels(
                this.props.labels,
                this.props.collapse,
                this.state.hover,
              ) :
              this.props.children &&
                (<StyledRightIcon
                  className="fa fa-angle-left"
                  open={this.state.open}
                  collapse={this.props.collapse}
                  hover={this.state.hover}
                />)
            )}
          </RightSpan>
        </StyledLink>
        {this.props.children && (
          <StyledSubMenu
            collapse={this.props.collapse}
            hover={this.state.hover}
            level={this.props.level}
            open={this.state.open}
          >
            {renderChildren(
              this.props.children,
              this.props.collapse,
              this.props.level,
              (this.state.hover || this.props.parentHover),
            )}
          </StyledSubMenu>
        )}
      </StyledMenuItem>
    );
  }
}

MenuItem.propTypes = {
  children: React.PropTypes.node,
  active: React.PropTypes.bool,
  collapse: React.PropTypes.bool,
  icon: React.PropTypes.objectOf(React.PropTypes.string),
  href: React.PropTypes.string,
  labels: React.PropTypes.arrayOf(React.PropTypes.object),
  level: React.PropTypes.number,
  onClick: React.PropTypes.func,
  parentHover: React.PropTypes.bool,
  title: React.PropTypes.string,
};

MenuItem.defaultProps = {
  collapse: false,
  icon: { className: 'fa-circle-o' },
  href: null,
  level: 0,
  parentHover: false,
  title: 'Title',
};

export default MenuItem;
