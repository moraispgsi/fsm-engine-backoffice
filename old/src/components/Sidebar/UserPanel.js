/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import {
  fontFamilyBase,
  fontSizeBase,
  fontWeightBase,
} from '../../styles/variables';

const StyledIcon = styled.i`
  margin-right: 3px;
  color: ${props => (props.online ? '#3c763d' : '#a94442')};
`;

const StyledP = styled.p`
  margin: 0 0 9px 0;
  font-weight: 600;
`;

const StyledLink = styled.a`
  text-decoration: none !important;
  padding-right: 5px;
  margin-top: 3px;
  font-size: 11px;
  background-color: transparent;
  color: ${props => props.theme.userPanelColor || '#fff'} !important;

  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */
`;

const StyledInfo = styled.div`
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  color: ${props => props.theme.userPanelColor || '#fff'};
  display: ${props => (props.collapse ? 'none !important' : 'block')};
  float: left !important;
  padding: 5px 5px 5px 15px;
  line-height: 1;
  position: absolute;
  left: 55px;

  /* collapse transform */
  ${props => props.collapse && '-webkit-transform: translateZ(0);'}
`;

const StyledImage = styled.img`
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  display: block;
  float: left!important;
  width: 100%;
  max-width: 45px;
  vertical-align: middle;
  height: auto;
  border-radius: 50%;
  border: 0;

  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */
`;

const StyledPanel = styled.div`
  /* clearfix */
  &:before, &:after {
    display: table;
    content: " ";
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
  &:after {
    clear: both;
  }

  /* shared */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: ${fontFamilyBase};
  font-weight: ${fontWeightBase};
  font-size: ${fontSizeBase};
  box-sizing: border-box;

  position: relative;
  width: 100%;
  padding: 10px;
  overflow: hidden;
  display: block;
  white-space: nowrap;
`;

const UserPanel = ({ name, image, href = null, online, collapse }) => (
  <StyledPanel collapse={collapse} >
    <StyledImage src={image} />
    <StyledInfo collapse={collapse} >
      <StyledP>{name}</StyledP>
      <StyledLink href={href}>
        <StyledIcon className="fa fa-circle" online={online} />
        {online ? ' Online' : ' Offline'}
      </StyledLink>
    </StyledInfo>
  </StyledPanel>
);

UserPanel.propTypes = {
  name: React.PropTypes.string,
  image: React.PropTypes.string,
  href: React.PropTypes.string,
  online: React.PropTypes.bool,
  collapse: React.PropTypes.bool,
};

export default UserPanel;
