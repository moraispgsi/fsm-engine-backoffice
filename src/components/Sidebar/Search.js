/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import {
  fontFamilyBase,
  fontSizeBase,
  fontWeightBase,
  lineHeightBase,
} from '../../styles/variables';

const StyledIcon = styled.i`
  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */
`;

const StyledButton = styled.button`
  z-index: 2;
  margin-left: -1px;
  font-size: ${fontSizeBase};
  padding: 6px 12px;
  background-image: none;
  font-family: ${fontFamilyBase};
  line-height: ${lineHeightBase};
  overflow: visible;
  position: relative;
  cursor: pointer;
  vertical-align: middle;
  touch-action: manipulation;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */

  /* theme */
  color: ${props => props.theme.sidebarFormButtonColor};
  border-radius: ${props => props.theme.sidebarFormButtonBorderRadius};
  box-shadow: ${props => props.theme.sidebarFormInputBoxShadow};
  background-color: ${props => props.theme.sidebarFormInputBg};
  border: ${props => props.theme.sidebarFormInputBorder};
  height: ${props => props.theme.sidebarFormInputHeight};
`;

const StyledInputButton = styled.span`
  position: relative;
  font-size: 0;
  white-space: nowrap;
  width: 1%;
  vertical-align: middle;
  display: table-cell;
`;

const StyledInput = styled.input`
  -webkit-writing-mode: horizontal-tb;
  -webkit-rtl-ordering: logical;
  user-select: text;
  cursor: auto;
  text-rendering: auto;
  letter-spacing: normal;
  word-spacing: normal;
  text-transform: none;
  text-indent: 0px;
  text-shadow: none;
  text-align: start;
  background-image: none;

  margin: 0;
  font: inherit;
  font-family: inherit;

  /* form-control */
  padding: 6px 12px;
  -webkit-transition: border-color ease-in-out .15s,-webkit-box-shadow ease-in-out .15s;
  -o-transition: border-color ease-in-out .15s,box-shadow ease-in-out .15s;
  transition: border-color ease-in-out .15s,box-shadow ease-in-out .15s;

  &:not(select) {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
  }

  position: relative;
  z-index: 2;
  float: left;
  width: 100%;
  margin-bottom: 0;
  display: table-cell;

  /* theme */
  box-shadow: ${props => props.theme.sidebarFormInputBoxShadow};
  background-color: ${props => props.theme.sidebarFormInputBg};
  border: ${props => props.theme.sidebarFormInputBorder};
  height: ${props => props.theme.sidebarFormInputHeight};
  color: ${props => props.theme.sidebarFormInputColor};
  border-radius: ${props => props.theme.sidebarFormInputBorderRadius};

  &:focus {
    background-color: ${props => props.theme.sidebarFormInputFocusBg};
    color: ${props => props.theme.sidebarFormInputFocusColor};
    border-left-color: ${props => props.theme.sidebarFormInputFocusBorderLeftColor};
  }
`;

const InputGroup = styled.div`
  position: relative;
  display: table;
  border-collapse: separate;

  &:focus { outline: -webkit-focus-ring-color auto 5px; }
`;

const StyledForm = styled.form`
  /* shared */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: ${fontFamilyBase};
  font-weight: ${fontWeightBase};
  font-size: ${fontSizeBase};
  line-height: ${lineHeightBase};
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;

  overflow: hidden;
  text-overflow: clip;
  border-radius: ${props => props.theme.sidebarFormBorderRadius || '0'};
  border: ${props => props.theme.sidebarFormBorder || '0'};
  margin: ${props => props.theme.sidebarFormMargin || '0'};

  /* collapse */
  ${props => props.collapse && `
    display: none !important;
    -webkit-transform: translateZ(0);
  `}
`;

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
    };
    this._changeValue = this.changeValue.bind(this);
    this._buttonClick = this.buttonClick.bind(this);
  }

  changeValue(event) {
    this.setState({ value: event.target.value });
  }

  buttonClick() {
    this.props.onClick(this.state.value);
    this.setState({ value: '' });
  }

  render() {
    return (
      <StyledForm collapse={this.props.collapse} >
        <InputGroup>
          <StyledInput
            type="text"
            name={this.props.name}
            placeholder={this.props.placeholder}
            value={this.state.value}
            onChange={this._changeValue}
          />
          <StyledInputButton>
            <StyledButton name="searchButton" onClick={this._buttonClick}>
              <StyledIcon className="fa fa-search" />
            </StyledButton>
          </StyledInputButton>
        </InputGroup>
      </StyledForm>
    );
  }
}

Search.propTypes = {
  name: React.PropTypes.string,
  placeholder: React.PropTypes.string,
  onClick: React.PropTypes.func,
  collapse: React.PropTypes.bool,
};

Search.defaultProps = {
  placeholder: 'Search...',
  onClick: v => alert(`Searching for ${v}`),
  collapse: false,
};

export default Search;
