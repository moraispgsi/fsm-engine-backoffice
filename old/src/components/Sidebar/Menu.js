/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import MenuItem from './MenuItem';

const StyledHeader = styled.li`
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  display: ${props => (props.collapse ? 'none !important' : 'list-item')};
  text-align: -webkit-match-parent;
  position: relative;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  padding: 10px 25px 10px 15px;
  font-size: 12px;
  background-attachment: scroll;
  background-clip: initial;
  background-image: none;
  background-origin: initial;
  background-position: initial;
  background-repeat: initial;
  background-size: initial;
  text-overflow: clip;

  /* collapse transform */
  ${props => props.collapse && '-webkit-transform: translateZ(0);'}

  /* theme */
  color: ${props => props.theme.sidebarHeaderColor || '#fff'}
  background-color: ${props => props.theme.sidebarHeaderBg || '#fff'}
`;

const StyledMenu = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  overflow: hidden;

  &:hover {
    overflow: visible;
  }
`;

const renderChildren = (children, sidebarCollapse) => (
  React.Children.map(children, child =>
    React.cloneElement(child, {
      collapse: sidebarCollapse,
    }),
  )
);

const Menu = ({ children, header, collapse }) => (
  <StyledMenu collapse={collapse} >
    <StyledHeader collapse={collapse} >{header}</StyledHeader>
    {renderChildren(children, collapse)}
  </StyledMenu>
);

Menu.propTypes = {
  children: React.PropTypes.node,
  header: React.PropTypes.string,
  collapse: React.PropTypes.bool,
};

Menu.Item = MenuItem;

export default Menu;
