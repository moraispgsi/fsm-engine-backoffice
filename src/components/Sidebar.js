/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import UserPanel from './Sidebar/UserPanel';
import Search from './Sidebar/Search';
import Menu from './Sidebar/Menu';

import {
  screenXsMax,
  screenSmMin,
  sidebarWidth,
  sidebarMiniWidth,
  transitionSpeed,
  transitionFn,
} from '../styles/variables';

const StyledSection = styled.section`
  height: auto;
  padding-bottom: 10px;
  display: block;
`;

const StyledAside = styled.aside`
  &:before, &:after {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }

  /* theme */
  background-color: ${props => props.theme.sidebarBg}

  position: ${props => (props.fixed ? 'fixed' : 'absolute')};
  top: 0;
  left: 0;
  padding-top: 50px;
  min-height: 100vh;
  height: 100%;
  width: ${sidebarWidth};
  z-index: 810;

  -webkit-transition: ${transitionSpeed} ${transitionFn}, width ${transitionSpeed} ${transitionFn};
  -moz-transition: ${transitionSpeed} ${transitionFn}, width ${transitionSpeed} ${transitionFn};
  -o-transition: ${transitionSpeed} ${transitionFn}, width ${transitionSpeed} ${transitionFn};
  transition: ${transitionSpeed} ${transitionFn}, width ${transitionSpeed} ${transitionFn};

  /* media queries */
  /* sidebar on large screens */
  @media (min-width: ${screenSmMin}) {
    margin-left: 0;
    ${(props) => {
      if (props.collapse) {
        if (props.mini) {
          return `
            width: ${sidebarMiniWidth};
            z-index: 850;
          `;
        }
        return `
          -webkit-transform: translate(-${sidebarWidth}, 0);
          -ms-transform: translate(-${sidebarWidth}, 0);
          -o-transform: translate(-${sidebarWidth}, 0);
          transform: translate(-${sidebarWidth}, 0);
        `;
      }
      return `
        -webkit-transform: translate(0, 0);
        -ms-transform: translate(0, 0);
        -o-transform: translate(0, 0);
        transform: translate(0, 0);
      `;
    }}
  }
  /* sidebar on small screens */
  @media (max-width: ${screenXsMax}) {
    padding-top: 100px;
    ${props => (props.collapse ? `
      -webkit-transform: translate(-${sidebarWidth}, 0);
      -ms-transform: translate(-${sidebarWidth}, 0);
      -o-transform: translate(-${sidebarWidth}, 0);
      transform: translate(-${sidebarWidth}, 0);
    ` : `
      -webkit-transform: translate(0, 0);
      -ms-transform: translate(0, 0);
      -o-transform: translate(0, 0);
      transform: translate(0, 0);
    `)};
  }
`;

const renderChildren = (children, sidebarCollapse) => (
  React.Children.map(children, child =>
    React.cloneElement(child, {
      collapse: sidebarCollapse,
    }),
  )
);

const Sidebar = ({
  children,
  fixed = false,
  sidebarMini = false,
  sidebarCollapse = false,
}) => (
  <StyledAside
    fixed={fixed}
    collapse={sidebarCollapse}
    mini={sidebarMini}
  >
    <StyledSection name="sidebar-wrapper">
      {renderChildren(children, sidebarCollapse)}
    </StyledSection>
  </StyledAside>
);

Sidebar.propTypes = {
  children: React.PropTypes.node,
  fixed: React.PropTypes.bool,
  sidebarMini: React.PropTypes.bool,
  sidebarCollapse: React.PropTypes.bool,
};

Sidebar.UserPanel = UserPanel;
Sidebar.Search = Search;
Sidebar.Menu = Menu;

export default Sidebar;
