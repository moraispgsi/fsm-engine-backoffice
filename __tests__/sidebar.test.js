/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import red from '../src/styles/skin-red';
import Sidebar from '../src/components/Sidebar';

describe('<Sidebar />', () => {
  it('renders correctly with default options', () => {
    const component = renderer.create(<Sidebar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with theme', () => {
    const component = renderer.create(<ThemeProvider theme={red}>
      <Sidebar />
    </ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed option', () => {
    const component = renderer.create(<Sidebar fixed />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with collapsed option', () => {
    const component = renderer.create(<Sidebar sidebarCollapse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed and collapsed options', () => {
    const component = renderer.create(
      <Sidebar fixed sidebarCollapse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with mini option', () => {
    const component = renderer.create(<Sidebar sidebarMini />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with mini and fixed options', () => {
    const component = renderer.create(<Sidebar sidebarMini fixed />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with mini and collapsed options', () => {
    const component = renderer.create(
      <Sidebar sidebarMini sidebarCollapse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with mini, fixed, and collapsed options', () => {
    const component = renderer.create(
      <Sidebar sidebarMini fixed sidebarCollapse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const component = renderer.create(<Sidebar >
      <span>Child</span>
    </Sidebar>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
