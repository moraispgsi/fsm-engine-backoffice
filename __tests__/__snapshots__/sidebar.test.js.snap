exports[`<Sidebar /> renders children correctly 1`] = `
<aside
  className="MmKeu">
  <section
    className="hZXMYD"
    name="sidebar-wrapper">
    <span
      collapse={false}>
      Child
    </span>
  </section>
</aside>
`;

exports[`<Sidebar /> renders correctly with collapsed option 1`] = `
<aside
  className="isrQCO">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with default options 1`] = `
<aside
  className="MmKeu">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with fixed and collapsed options 1`] = `
<aside
  className="hPSzSq">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with fixed option 1`] = `
<aside
  className="gyIrIH">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with mini and collapsed options 1`] = `
<aside
  className="kHCXGh">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with mini and fixed options 1`] = `
<aside
  className="gyIrIH">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with mini option 1`] = `
<aside
  className="MmKeu">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with mini, fixed, and collapsed options 1`] = `
<aside
  className="hbjQrz">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;

exports[`<Sidebar /> renders correctly with theme 1`] = `
<aside
  className="dDcugc">
  <section
    className="hZXMYD"
    name="sidebar-wrapper" />
</aside>
`;
